pkg/controller/cincinnati/new.go

```diff
@@ -333,36 +321,105 @@ func (k *kubeResources) newDeployment(instance *cv1beta1.Cincinnati) *appsv1.Dep
 	return dep
 }
 
-func (k *kubeResources) addExternalCACert(instance *cv1beta1.Cincinnati) {
+func (k *kubeResources) newVolumes(instance *cv1beta1.Cincinnati) *[]corev1.Volume {
+	mode := int32(420) // 0644
+	return &[]corev1.Volume{
+		corev1.Volume{
+			Name: "configs",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					DefaultMode: &mode,
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: nameConfig(instance),
+					},
+				},
+			},
+		},
+		corev1.Volume{
+			Name: "cincinnati-graph-data",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+}
+
+func (k *kubeResources) newTrustedCAConfig(instance *cv1beta1.Cincinnati, cm *corev1.ConfigMap) {
+	k.trustedCAConfig = &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nameAdditionalTrustedCA(instance),
+			Namespace: instance.Namespace,
+		},
+		Data: cm.Data,
+	}
+}
+
+func (k *kubeResources) newPullSecret(instance *cv1beta1.Cincinnati, s *corev1.Secret) {
+	k.pullSecret = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namePullSecretCopy(instance),
+			Namespace: instance.Namespace,
+		},
+		Data: s.Data,
+	}
+}
+
+// regenerate - rebuild smaller resources to regerate the larger object
```

Comment:
s/regerate/regenerate/

Reply:
I'm generally not sure why this function is necessary. What is it about the container and the deployment that need to be regenerated?

Reply:
Since we're modifying the graph builder container object and not the deployment, the deployment needs to be regenerated to pickup the changes.  I made two changes on that concept:
1) I pulled the regeneration logic out of `ensureAdditionalTrustedCA` to make it visible as the final step in the post resource modification process. 
2) I broke the objects I'm modifying down to their smallest form to more easily manipulate them.  Then, I rebuild each larger object one by one, which I found it to be less error prone since it only requires one call to `newDeployment`.

For example:
1) Append the pull-secret volumeMount to the graphBuilderVolumeMounts
2) Rebuild the graphBuilderContainer object
3) Rebuild the deployment